doc/ref/states/requisites.rst

```diff
@@ -423,6 +512,43 @@ The ``onfail`` requisite is applied in the same way as ``require`` as ``watch``:
 
 .. _Issue #22370: https://github.com/saltstack/salt/issues/22370
 
+.. _requisites-onfail:
+
+onfail_any
+~~~~~~~~~~
+
+.. versionadded:: Oxygen
+
+The ``onfail_any`` requisite allows for reactions to happen strictly as a response
+to the failure of at least one other state. This can be used in a number of ways, such as
+executing a second attempt to set up a service or begin to execute a separate
+thread of states because of a failure.
+
+The ``onfail_any`` requisite is applied in the same way as ``require_any`` as ``watch_any``:
+
+.. code-block:: yaml
+
+    primary_mount:
+      mount.mounted:
+        - name: /mnt/share
+        - device: 10.0.0.45:/share
+        - fstype: nfs
+
+    secondary_mount:
+      mount.mounted:
+        - name: /mnt/code
+        - device: 10.0.0.45:/code
+        - fstype: nfs
+
+    backup_mount:
+      mount.mounted:
+        - name: /mnt/share
+        - device: 192.168.40.34:/share
+        - fstype: nfs
+        - onfail:
+          - mount: primary_mount
+          - mount: secondary_mount
+
```

Comment:
Can you add a sentence here about what the expected outcome of the state run will be, as you've done for the other examples so far?